pkg/test/e2e/default_app_catalog_tests/application_test.go

```diff
@@ -0,0 +1,408 @@
+		if err != nil {
+			return fmt.Errorf("failed to verify that helm release is deployed: %w", err), nil
+		}
+
+		return nil, nil
+	})
+	if err != nil {
+		t.Fatalf("Application observe test failed: %v", err)
+	}
+
+	log.Info("Checking if all conditions are ok")
+	err = wait.PollLog(ctx, log, 2*time.Second, 5*time.Minute, func(ctx context.Context) (error, error) {
+		err = checkApplicationInstallationConditions(ctx, log, client)
+		if err != nil {
+			return fmt.Errorf("failed to verify that all conditions are in healthy state: %w", err), nil
+		}
+
+		return nil, nil
+	})
+	if err != nil {
+		t.Fatalf("Application observe test failed: %v", err)
+	}
+
+	namesStrArray := strings.Split(names, ",")
+	log.Info("Waiting for pods to get ready...")
+	err = wait.PollLog(ctx, log, 2*time.Second, 7*time.Minute, func(ctx context.Context) (error, error) {
+		err = waitForPods(ctx, log, client, applicationNamespace, key, namesStrArray)
+		if err != nil {
+			return fmt.Errorf("failed to verify that all pods are ready: %w", err), nil
+		}
+
+		return nil, nil
+	})
+	if err != nil {
+		t.Fatalf("Application observe test failed: %v", err)
+	}
+}
+
+func waitForPods(ctx context.Context, log *zap.SugaredLogger, client ctrlruntimeclient.Client, namespace string, key string, names []string) error {
+	log = log.With("namespace", namespace)
+
+	r, err := labels.NewRequirement(key, selection.In, names)
+	if err != nil {
+		return fmt.Errorf("failed to build requirement: %w", err)
+	}
+	l := labels.NewSelector().Add(*r)
+
+	return wait.PollLog(ctx, log, 5*time.Second, 5*time.Minute, func(ctx context.Context) (error, error) {
+		pods := corev1.PodList{}
+		err = client.List(ctx, &pods, ctrlruntimeclient.InNamespace(namespace), ctrlruntimeclient.MatchingLabelsSelector{Selector: l})
+		if err != nil {
+			return fmt.Errorf("failed to list Pods: %w", err), nil
+		}
+
+		if len(pods.Items) == 0 {
+			return errors.New("no Pods found"), nil
+		}
+
+		unready := sets.New[string]()
+		for _, pod := range pods.Items {
+			ready := false
+			for _, c := range pod.Status.Conditions {
+				if c.Type == corev1.ContainersReady {
+					ready = c.Status == corev1.ConditionTrue
+				}
+			}
+
+			if !ready {
+				unready.Insert(pod.Name)
+			}
+		}
+
+		if unready.Len() > 0 {
+			return fmt.Errorf("not all Pods are ready: %v", sets.List(unready)), nil
+		}
+
+		return nil, nil
+	})
+}
+
+func isHelmReleaseDeployed(ctx context.Context, log *zap.SugaredLogger, client ctrlruntimeclient.Client, appName, namespace string) error {
+	secrets := corev1.SecretList{}
+	err := client.List(ctx, &secrets, ctrlruntimeclient.InNamespace(namespace))
+	if err != nil {
+		log.Fatalf("failed to list secrets: %v", err)
+		return err
+	}
+
+	for _, secret := range secrets.Items {
+		if containsString(secret.Name, appName) && secret.Type == "helm.sh/release.v1" {
+			if status, exists := secret.Labels["status"]; exists && status == "deployed" {
```

Comment:
since it is a string[string] map, it is safe to directly evaluate it like this: the value will be "" if the map entry does not exist


```suggestion
			if secret.Labels["status"] == "deployed" {
```